hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/acm/AccessControlManager.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2023 Bosch.IO GmbH and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hawkbit.repository.jpa.acm;
+
+import org.eclipse.hawkbit.repository.exception.InsufficientPermissionException;
+import org.eclipse.hawkbit.repository.jpa.autoassign.AutoAssignChecker;
+import org.eclipse.hawkbit.repository.jpa.model.JpaTargetFilterQuery;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Interface of an access control manager. Used by management layer to verify
+ * the permission for CRUD operations based on some access criteria.
+ * 
+ * @param <T>
+ */
+public interface AccessControlManager<T> {
```

Comment:
maybe AccessControl(ler) as in Java security would be more appropriate and concise?
it's not something that manages control but applies it